theano/compile/nanguardmode.py

```diff
@@ -239,20 +271,24 @@ def do_check_on(var, nd, f, is_input):
                     logger.error('Big value detected')
                     error = True
             if error:
-                if is_input:
-                    logger.error('In an input')
+                if not is_input:
+                    logger.error("NanGuardMode found an error in the"
+                                 " output of a node in this variable:")
+                    logger.error(theano.printing.debugprint(nd, file='str'))
                 else:
-                    logger.error('In an output')
-                logger.error('Inputs: ')
-                for ivar, ival in zip(nd.inputs, f.inputs):
-                    logger.error('var')
-                    logger.error(ivar)
-                    logger.error(theano.printing.min_informative_str(ivar))
-                    logger.error('val')
-                    logger.error(ival)
-                logger.error('Node:')
-                logger.error(nd)
-                assert False
+                    logger.error("NanGuardMode found an error in the"
+                                 " input %d of this node.")
+                    logger.error('Node:')
+                    logger.error(nd)
+                    logger.error("The input variable that cause problem:")
+                    logger.error(theano.printing.debugprint(nd, file='str'))
+                if config.NanGuardMode.action == 'raise':
+                    assert False
```

Comment:
Raising some exception with the error message that is logged above is what I would expect for `"raise"`